projects/core/src/occ/services/occ-fields.service.ts

```diff
@@ -0,0 +1,179 @@
+import { Injectable } from '@angular/core';
+import { extractFields, mergeFields, parseFields } from '../utils/occ-fields';
+import { ScopedData } from '../../model/scoped-data';
+import { map, shareReplay } from 'rxjs/operators';
+import { Observable } from 'rxjs';
+import { HttpClient } from '@angular/common/http';
+
+/**
+ * Intermediate model to accommodate all data needed to perform occ fields optimizations
+ * wrapping ScopedData with url and fields
+ */
+export interface OccFieldsModels {
+  /** Url (with fields) to load scoped data */
+  url?: string;
+  /** extracted fields object, used to extract data from broader model */
+  fields?: object;
+  /** scoped data model */
+  model: ScopedData<any>;
```

Comment:
1. Agreed, renamed.
2. It's hard to do it, because it's intermediate type and using it in other methods is not straightforward (we would have to create one more type to make it strict enough). Also, we don't care about a specific types of scoped data in this logic, that's why `any` is reasonable IMO.